### ISSUE TITLE
Useless lifetimes

### ISSUE THREAD

**Issue by Bowarc** [sardonicism-04/discord-rich-presence#19]
Hellow, im tring to have some sort of dynamic Activity,

**Comment by antiaim** [comment]
Agreed, was trying to run the ipc client in a thread but couldnt share the Activity struct with the thread because of the lifetimes. ended up having to just make another struct that contains the data and sharing that instead.

**Comment by Bowarc** [comment]
Yes, why not.
Having functions like 
```rust 
Activity::set_state(&mut self, new_state: &str) -> () {
    self.state=new_state
}
```
Could be very helpfull for thoses kind of senarios.

**Comment by Bowarc** [comment]
I tried a bit more, and it turns out that, in my case, using the kw `ref` in the `Some()` section of the `match` compiles.
`
match activity_config.state {
    Some(ref state) => {
        debug!("Activity::State set to: {}", state);
        activity = activity.state(&state);
    }
    None => debug!("Activity::State not found"),
}
match activity_config.details {
    Some(ref details) => {
        debug!("Activity::Details set to: {}", details);
        activity = activity.details(&details);
    }
    None => debug!("Activity::Details not found"),
}
`
Even if that case worked; for other scenarios, having functions that take `&mut self` would be greatly appreciated.
